src/core/crypto.c

```diff
@@ -2255,16 +2625,23 @@ QuicCryptoEncodeServerTicket(
     //
 
     _Analysis_assume_(sizeof(*TicketBuffer) >= 8);
-    uint8_t* TicketCursor = QuicVarIntEncode(CXPLAT_TLS_RESUMPTION_TICKET_VERSION, TicketBuffer);
+    uint8_t* TicketCursor = QuicVarIntEncode(CXPLAT_TLS_RESUMPTION_TICKET_MAX_VERSION, TicketBuffer);
     CxPlatCopyMemory(TicketCursor, &QuicVersion, sizeof(QuicVersion));
     TicketCursor += sizeof(QuicVersion);
     TicketCursor = QuicVarIntEncode(AlpnLength, TicketCursor);
     TicketCursor = QuicVarIntEncode(EncodedTPLength, TicketCursor);
+    TicketCursor = QuicVarIntEncode(EncodedCRLength, TicketCursor);
     TicketCursor = QuicVarIntEncode(AppDataLength, TicketCursor);
     CxPlatCopyMemory(TicketCursor, NegotiatedAlpn, AlpnLength);
     TicketCursor += AlpnLength;
     CxPlatCopyMemory(TicketCursor, EncodedHSTP + CxPlatTlsTPHeaderSize, EncodedTPLength);
     TicketCursor += EncodedTPLength;
+
+    if (NULL != CarefulResumeState) {
+        QuicCryptoEncodeCRState(EncodedCRLength, CarefulResumeState, Connection, &EncodedCRLength, TicketCursor);
```

Comment:
Assuming the comment above has been fixed, on failure, `EncodedCRLength == 0`.
So `*TicketLength = TotalTicketLength` won't be correct anymore.

Reply:
I don't understand why we have an out parameter in this function for the size if the function assumes it receive the correct amount of memory. Either the function assumes the caller provide a valid buffer and doesn't need to check it, or it doesn't, and it must handle failure.